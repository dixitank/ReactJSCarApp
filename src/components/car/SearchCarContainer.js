import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import toastr from 'toastr';
import * as carAction from '../../action/CarAction';
import CarSearchForm from './CarSearchForm'; 


export class SearchCarContainer extends React.Component {


    constructor() {
        super();
        this.handleSave = this.handleSave.bind(this);
        this.handleCancel = this.handleCancel.bind(this);
    }



    componentDidMount() {

    }



    handleSave(values) {
        let distanceToTravel = parseInt(values.distance);
        let fuelPricePerLitre = parseFloat(values.cost);

        this.props.action.getCarsByMileageAction(distanceToTravel, fuelPricePerLitre)
            .then(() => {
                this.props.history.push('/viewCarsByCustomSearch');
            }).catch(error => {
                toastr.error(error);
            });
    }



    handleCancel(event) {
        event.preventDefault();
        this.props.history.replace('/carsByMileage');
    }



    render() {
        const { initialValues } = this.props;
        const heading = initialValues && initialValues.id ? 'Edit' : 'Add';

        return (
            <div className="container">
                 <CarSearchForm
                    heading={heading}  
                    handleSave={this.handleSave}
                    handleCancel={this.handleCancel}
                    initialValues={this.props.initialValues}
                />
            </div>
        );
    }
}



const mapStateToProps = (state, ownProps) => {
   return {}
};



const mapDispatchToProps = dispatch => ({
    action: bindActionCreators({ ...carAction }, dispatch)
});



SearchCarContainer.propTypes = {
    action: PropTypes.object.isRequired,
    history: PropTypes.object,
    initialValues: PropTypes.object,
    match: PropTypes.object.isRequired
};



export default connect(mapStateToProps, mapDispatchToProps)(SearchCarContainer);
