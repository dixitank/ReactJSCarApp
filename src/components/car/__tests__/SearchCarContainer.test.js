import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {SearchCarContainer} from '../SearchCarContainer';


describe('SearchCarContainer.test.js', () => {

    it('renders without crashing', () => {
        const props = {
            action: {
                getCarAction: jest.fn()
            },
            initialValues: {brand: 'Fusion'},
            match: {params: {id:'1'}}
        };

        const wrapper = shallow(<SearchCarContainer {...props}/>);
        expect(wrapper.length).toEqual(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

});



    
        