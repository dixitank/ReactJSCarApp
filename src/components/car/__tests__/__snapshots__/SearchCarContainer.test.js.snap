exports[`SearchCarContainer.test.js renders without crashing 1`] = `
<div
  className="container">
  <ReduxForm
    handleCancel={[Function]}
    handleSave={[Function]}
    heading="Add"
    initialValues={
      Object {
        "brand": "Fusion",
      }
    } />
</div>
`;
